diff -ruP -x imp -x '*.webprj' horde-2.1/imp-3.1/config/prefs.php horde-2.1-patched/imp-3.1/config/prefs.php
--- horde-2.1/imp-3.1/config/prefs.php	Fri Jul  5 09:51:23 2002
+++ horde-2.1-patched/imp-3.1/config/prefs.php	Wed Aug 28 22:22:06 2002
@@ -19,6 +19,13 @@
 // The $_prefs array's are listed in the same order as listed in the
 // members element of $prefGroups

+$prefGroups['pbWebMAUI'] = array(
+    'column' => _("Mailserver"),
+    'label' => _("Mailaccount"),
+    'desc' => _("Ändern Sie Mailserver Einstellungen zu Ihrem Mailkonto.<br>Ändern Sie z.B. Ihr Kennwort, Filterregeln oder Abwesenheitsnachrichten.<br><br>Dies ist der integrierte Aufruf von pb.WebMAUI"),
+    'members' => array('pbWebMAUI')
+);
+
 $prefGroups['personal'] = array(
     'column' => _("Your Information"),
     'label' => _("Personal Information"),
@@ -147,6 +154,8 @@
 //   enum: Static list of elements.
 //

+//pb.WebMAUI
+$_prefs['pbWebMAUI'] = array('type' => 'special');

 // Personal Information preferences

diff -ruP -x imp -x '*.webprj' horde-2.1/imp-3.1/templates/prefs/pbWebMAUI.inc horde-2.1-patched/imp-3.1/templates/prefs/pbWebMAUI.inc
--- horde-2.1/imp-3.1/templates/prefs/pbWebMAUI.inc	Thu Jan  1 01:00:00 1970
+++ horde-2.1-patched/imp-3.1/templates/prefs/pbWebMAUI.inc	Wed Aug 28 23:00:43 2002
@@ -0,0 +1,36 @@
+<?php
+/* this would be nice to work without iframe, but we have to send get-parms somehow to pb.WebMAUI
+    and pbWebMAUI has to create horde links...
+
+    so for now let's stick with an iframe
+
+    $file = fopen("http://localhost/pb.WebMAUI/", "r");
+    if (!$file) {
+        echo "<p>Unable to open remote file.\n";
+        exit;
+    }
+    while (!feof($file)) {
+        $line = fgets($file, 1024);
+        echo $line;
+    }
+    fclose($file);
+*/
+    $uri = "/pb.WebMAUI/index.php?style=imp";
+    $uri .= "&user=".$imp['user'];
+    $uri .= "&password=".Secret::read(Secret::getKey('imp'), $imp['pass']);
+    $uri .= "&action=EditAccount";
+    $uri .= "&mail=".urlencode($imp['user']);
+?>
+<iframe src="<?= $uri ?>" frameborder="0" width="100%" height="450" name="pbWebMAUI_in_a_box">
+<p>Ihr Browser kann leider keine eingebetteten Frames anzeigen:
+Sie k&ouml;nnen die eingebettete Seite &uuml;ber den folgenden Verweis
+aufrufen:
+<a href="<?= $uri ?>" target="pbWebMAUI_in_a_window">pb.WebMAUI</a></p>
+<script language="JavaScript" type="text/javascript">
+<!--
+window.open("<?= $uri ?>", "pbWebMAUI_in_a_window", "width=600,height=350");
+//-->
+</script>
+</iframe>
+</td></tr></table>
+<!-- uncomment following buttons from end.inc
\ No newline at end of file
diff -ruP -x imp -x '*.webprj' horde-2.1/templates/prefs/begin.inc horde-2.1-patched/templates/prefs/begin.inc
--- horde-2.1/templates/prefs/begin.inc	Mon Mar 11 17:57:53 2002
+++ horde-2.1-patched/templates/prefs/begin.inc	Mon Aug 26 13:37:30 2002
@@ -4,4 +4,4 @@

 <table border="0" align="center" width="100%" cellspacing="0" cellpadding="3">
 <tr><td class="header" align="left"><b><?= $prefGroups[$group]['label'] ?></b></td></tr>
-<tr><td class="item"><table border="0"><tr valign="top"><td class="item">
+<tr><td class="item"><table width="100%" border="0"><tr valign="top"><td class="item">
diff -ruP -x imp -x '*.webprj' horde-2.1/templates/prefs/end.inc horde-2.1-patched/templates/prefs/end.inc
--- horde-2.1/templates/prefs/end.inc	Sun Feb  3 18:44:32 2002
+++ horde-2.1-patched/templates/prefs/end.inc	Wed Aug 28 22:21:36 2002
@@ -5,5 +5,6 @@
 <input type="button" class="button" value="<?= _("Save Options") ?>" onclick="document.prefs.submit();" />&nbsp;
 <input type="reset" class="button" value="<?= _("Undo Changes") ?>" />&nbsp;
 <input type="button" class="button" value="<?= _("Return to Options") ?>" onclick="document.prefs.actionID.value=0; document.prefs.group.value=''; document.prefs.submit();" />
+<!-- used to uncomment buttons for external prefs -->
 </td></tr></table>
 </form>
